src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs

```diff
@@ -147,6 +147,12 @@ public string TestCode
         /// </summary>
         public List<Func<Solution, ProjectId, Solution>> SolutionTransforms { get; } = new List<Func<Solution, ProjectId, Solution>>();
 
+        /// <summary>
+        /// Gets or sets the timeout to use when matching expected and actual diagnostics. The default value is 2
+        /// seconds.
+        /// </summary>
+        protected TimeSpan MatchDiagnosticsTimeout { get; set; } = TimeSpan.FromSeconds(2);
```

Comment:
How was this decided? 2 seconds seems _very_ long to me.

Reply:
This has only been hit a couple times since the feature was implemented, and in both cases it was during the development of a new test. The timeout is mostly about providing the most usable error message in the event of a test failure, so they are naturally short-lived scenarios.